src/views/DatasetManager/components/CreateDatasetButton/components/DatasetCreationParameters/DatasetCreationParametersHook.js

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { useMemo, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { DATASET_SOURCE_TYPE } from '../../../../../../services/config/ApiConstants';
+import { useSolutionData } from '../../../../../../state/hooks/SolutionHooks';
+import { TranslationUtils, ConfigUtils } from '../../../../../../utils';
+
+export const useDatasetCreationParameters = () => {
+  const solutionData = useSolutionData();
+  const { t } = useTranslation();
+
+  // Each parameter must have a id strictly unique.
+  const hardCodedDataSources = useMemo(() => {
+    const dataSources = [
+      {
+        id: DATASET_SOURCE_TYPE.AZURE_STORAGE,
+        labels: {
+          en: 'Graph Format from Azure Storage',
+          fr: 'Format Graph depuis Azure Storage',
+        },
+        parameters: [
+          {
+            id: `name`,
+            varType: 'string',
+            labels: { en: 'Account name', fr: 'Nom du compte' },
+          },
+          {
+            id: `location`,
+            varType: 'string',
+            labels: { en: 'Container name', fr: 'Nom du container' },
+          },
+          {
+            id: `path`,
+            varType: 'string',
+            labels: { en: 'Path', fr: 'Chemin' },
+          },
+        ],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.ADT,
+        labels: {
+          en: 'Azure Digital Twin',
+          fr: 'Azure Digital Twin',
+        },
+        parameters: [
+          {
+            id: `location`,
+            varType: 'string',
+            labels: { en: 'Path', fr: 'Chemin' },
+          },
+        ],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.LOCAL_FILE,
+        labels: {
+          en: 'Graph Format from Local File',
+          fr: 'Format Graph depuis un fichier local',
+        },
+        parameters: [{ id: `file`, varType: '%DATASETID%', labels: { en: '', fr: '' } }],
+      },
+      {
+        id: DATASET_SOURCE_TYPE.NONE,
+        labels: {
+          en: 'Empty',
+          fr: 'Dataset vide',
+        },
+        parameters: [],
+      },
+    ];
+
+    dataSources.forEach((dataSource) => {
+      dataSource.parameters.forEach((parameter) => (parameter.id = `${dataSource.id}.${parameter.id}`));
+    });
+
+    TranslationUtils.addTranslationRunTemplateLabels(dataSources);
+    TranslationUtils.addTranslationParametersLabels(dataSources.flatMap((dataSource) => dataSource?.parameters));
+
+    return dataSources;
+  }, []);
+
+  const dataSourceRunTemplates = useMemo(() => {
+    const dataSources = solutionData.runTemplates.filter((runTemplate) => runTemplate?.tags.includes('datasource'));
```

Comment:
This line causes a crash of the webapp if a run template does not have any tag (`runTemplate.tags is undefined`)
It should probably be `runTemplate?.tags?.includes` (with the second optional chaining operator).